common/src/sql/dbinit.sql

```diff
@@ -669,15 +665,15 @@ CREATE TABLE omicron.public.identity_provider (
     provider_type omicron.public.provider_type NOT NULL
 );
 
-CREATE INDEX ON omicron.public.identity_provider (
-    id,
-    silo_id
+CREATE UNIQUE INDEX ON omicron.public.identity_provider (
```

Comment:
This one should already be unique because it contains the primary key.

I changed the order of the fields.  As it was, I don't think this index could be used to paginate through the identity providers in a Silo (which I think was the point of this index).